code/game/objects/structures/xeno.dm

```diff
@@ -750,9 +761,13 @@ TUNNEL
 		if(targettunnel && isturf(targettunnel.loc)) //Make sure the end tunnel is still there
 			M.forceMove(targettunnel)
 			var/double_check = input(M, "Emerge here?", "Tunnel: [targettunnel]") as null|anything in list("Yes","Pick another tunnel")
+			if(!targettunnel) //double check that the tunnel we're now in exists in the event it gets destroyed while we still have the interface open
+				return
+			if(M.loc != targettunnel) //double check that we're still in the tunnel in the event it gets destroyed while we still have the interface open
+				return
 			if(double_check == "Pick another tunnel")
 				return targettunnel.pick_a_tunnel(M)
-			else //Whether we say yes or cancel out of it
+			else //Whether we say yes or cancel out of it;
```

Comment:
??

Reply:
I think there was additional commentary I added that was later deleted